[Sajal] iP

SajalVaishnav opened the pull request:
#ChampakChacha2.0

Chacha ji once said:
> Arre babuchat JAA NAHANE JAA

With a champak chacha behind your back, no need to

* worry about forgetting your tasks,
* or deadlines,
* or events!!
* or bathing *913287498 times* a day

All you need to get your very own champak chacha is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you

And it is **FREE**

Features:

- [x] maintains your tasks 
- [x] reminds you of deadlines
- [ ] ghar mein bujurg laata h:family_man_man_girl_boy:
- [x] maintains your events 

If you know java you can even set the champak chacha dp. Change the following lines in `Duke.java`:
```
 private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));```

Commit:
Add Gradle support

SajalVaishnav committed:
greet, echo, exit w/o attitude

SajalVaishnav committed:
add, list functionality

SajalVaishnav committed:
mark as done

SajalVaishnav committed:
todos, events, deadlines

SajalVaishnav committed:
Automated Text UI Testing

SajalVaishnav committed:
added attitude and exception handling

SajalVaishnav committed:
delete, and more personality

SajalVaishnav committed:
package

SajalVaishnav committed:
file handling

SajalVaishnav committed:
Level-8 done

SajalVaishnav committed:
fixed some bugs

SajalVaishnav committed:
fixed bugs

SajalVaishnav committed:
Merge branch 'branch-Level-7'

SajalVaishnav committed:
Merge branch 'branch-Level-8'

SajalVaishnav committed:
More OOP

SajalVaishnav committed:
unit tests done

SajalVaishnav committed:
documentation

SajalVaishnav committed:
Merge branch 'branch-JAVAdocs'

SajalVaishnav committed:
coding standard

SajalVaishnav committed:
Merge branch 'branch-CodingStandard'

SajalVaishnav committed:
level-9 done

SajalVaishnav committed:
Merge branch 'branch-Level-9'

SajalVaishnav committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-GradleSupport

SajalVaishnav committed:
Merge branch 'branch-GradleSupport'

SajalVaishnav committed:
progress as of 11.30am 1/9/21

eeliana commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+
+    public String getTaskString() {
+        if(done) {
+            return "[D][X] " + this.taskString + " (by: " + this.deadline.format(DateTimeFormatter.ofPattern("MMM dd yyyy")) + ")";
+        } else {
+            return "[D][ ] " + this.taskString + " (by: " + this.deadline.format(DateTimeFormatter.ofPattern("MMM dd yyyy")) + ")";
```
Perhaps repeated code code can be extracted into a variable?

eeliana commented on src/main/java/duke/DeleteTaskMessage.java:
```diff
@@ -0,0 +1,20 @@
+        this.deletedTask = TaskList.getTaskList().getTasks().remove(taskIndex);
+    }
+    public void display() {
+        System.out.println("Theek h bhai... ye task hata diya");
+        System.out.println(deletedTask.getTaskString());
+        System.out.println("Ab " + TaskList.getTaskList().getSize() + " tasks bache list mein.");
```
Should these be in English?

SajalVaishnav replied:
Haha well yes... but prof said we can choose a personality and I chose an angry indian uncle... now there's no turning back

eeliana commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,29 @@
+ * The class for creating and managing Events added to TaskList. Extends Task.
+ */
+public class Event extends Task{
+    private boolean done;
```
Should the boolean be named as isDone instead?

eeliana commented on src/main/java/duke/FindMessage.java:
```diff
@@ -0,0 +1,35 @@
+        this.searchStr = searchStr;
+    }
+
+    public void display() {
+        ArrayList<Task> taskList = (ArrayList<Task>) TaskList.getTaskList().getTasks();
+        ArrayList<Task> searchResults = new ArrayList<>();
+        for(Task task: taskList) {
+            String taskStr = task.getTaskString();
+            if(taskStr.contains(searchStr)) {
+                searchResults.add(task);
+            }
+        }
+        int count = 1;
+        System.out.println("Here are the matching tasks in your list:\n");
+        for(Task task : searchResults) {
+            System.out.println(count + "." + task.getTaskString());
+            count = count + 1;
+        }
+    }
```
Maybe you can add a blank line of separation between the logical units within this block and possibly other places too?

eeliana commented on src/main/java/duke/TaskFactory.java:
```diff
@@ -0,0 +1,54 @@
+     * @exception EmptyDescriptionException If the description of a todo task is empty
+     * @exception InvalidCommandException If the taskStr doesn't match any task's format at all
+     */
+    public static Task createTask(String taskStr) throws IllegalFormatException, EmptyDescriptionException, InvalidCommandException {
```
Perhaps a line break here could improve readability? Does this line have more than 120 chars?

eeliana reviewed (commented):
Overall, some public methods and classes are missing documentation and maybe some areas where line breaks could improve readability. The code was generally not hard to read and understand but there are a few coding standard violations to fix. LGTM, after fixing these violations.

SajalVaishnav committed:
done with GUI

SajalVaishnav committed:
Merge branch 'branch-Level-10'

bnjmnt4n commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+ * The class for creating and managing Deadlines added to TaskList. Extends Task.
+ */
+public class Deadline extends Task {
+    private boolean done;
```
I think `done` can be renamed to `isDone` to better indicate that it is a boolean.

bnjmnt4n commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,24 @@
+ */
+public class Duke {
+    //true indicates the conversation is ongoing
+    public static boolean conversationState;
```
Do you think `conversationState` should be a publicly accessible static field which all the messages can access? I think that breaks the abstraction of the `Duke` class.

bnjmnt4n commented on src/main/java/duke/TaskFactory.java:
```diff
@@ -0,0 +1,54 @@
+            //create a todo task
+            try {
+                String taskDescription = taskStr.substring(5);
+                if(taskDescription.stripLeading().equals(""))
```
Should this be using Egyptian braces instead?

bnjmnt4n reviewed (commented):
The code is very readable, although there are some coding standards that are not met such as the Egyptian braces and variable naming. Also, I found the usage of `conversationState` a little confusing.

SajalVaishnav committed:
new JAR file

SajalVaishnav committed:
use assertions,remove redundant code,fix bugs

Currently,
* No assertions have been used to check for errors
* there is a redundant class
* data isn't being stored in local memory

Lets,
* use assertions to check for possible errors
* remove the redundant class
* fix the code so that duke stores the tasklist
  in memory

These are all standard practises, and that is enough
rationale.

SajalVaishnav committed:
Improve code quality

poor JAVA docs, some code style violations and one redundant class

* bad JAVA docs can make the code harder to debug and expand
* code style violations reduce the readablitity of the code
* redundant class also reduces readability

Lets,
* standardize JAVA docs
* fix code style violations
* remove the redundant class

Standardizing code style is a good way to improve code readability

SajalVaishnav committed:
Merge pull request #3 from SajalVaishnav/branch-A-CodeQuality

Improve code quality

SajalVaishnav committed:
Merge branch 'master' into branch-A-Assertions

SajalVaishnav committed:
allow search by date

the user couldn't search the tasks for a particular date

Lets add a parser to allow user to search tasks by date

This improves the product usability

SajalVaishnav committed:
Set theme jekyll-theme-merlot

SajalVaishnav committed:
added UI.png and updated user guide

SajalVaishnav committed:
Merge branch 'master' of https://github.com/SajalVaishnav/ip

SajalVaishnav committed:
improved code quality

SajalVaishnav committed:
improved code structure

SajalVaishnav committed:
improved user experience

SajalVaishnav committed:
removed a comment

SajalVaishnav committed:
fixed bug

SajalVaishnav committed:
fixed bugs

SajalVaishnav committed:
fixed bugs with loading from file

SajalVaishnav committed:
added some more JAVA docs

SajalVaishnav committed:
updated user guide

SajalVaishnav committed:
updated user guide 2.0

SajalVaishnav committed:
updated user guide 3.0

SajalVaishnav committed:
Set theme jekyll-theme-minimal

SajalVaishnav committed:
fixed storage bug

SajalVaishnav committed:
Merge branch 'master' of https://github.com/SajalVaishnav/ip

SajalVaishnav committed:
added translation for non-hindi speakers

SajalVaishnav committed:
added translation of greeting message
